library/ssl_tls13_server.c

```diff
@@ -1719,6 +1737,23 @@ int mbedtls_ssl_tls13_handshake_server_step( mbedtls_ssl_context *ssl )
             break;
 #endif /* MBEDTLS_KEY_EXCHANGE_WITH_CERT_ENABLED */
 
+        /*
+         * Injection of dummy-CCS's for middlebox compatibility
+         */
+#if defined(MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE)
+        case MBEDTLS_SSL_SERVER_CCS_AFTER_HELLO_RETRY_REUEST:
```

Comment:
Not sure whether we should add one state machine as this, but seems we have a way to call it at the beggining of next state machine(such as MBEDTLS_SSL_ENCRYPTED_EXTENSIONS)

Reply:
The client also using similar states for sending change_cipher_spec records.

Reply:
Let's keep one single message writing within the handler of writing states.